### ISSUE TITLE
Delegate submodule updates to the documentation's parent repo

### ISSUE THREAD

**Issue by rafrombrc** [readthedocs/readthedocs.org#529]
Commit 4215b8b3561840fc4959138a16ae0fe32fed8aa6 causes each submodule to be forcibly updated from master. This seems to me a Very Bad Idea™, b/c it's entirely possible for a submodule to be explicitly wired to a branch other than master.

Git submodule support is awkward at times, but even so the idea is that the parent repo has control over what revision is used for each submodule. If a parent module wants to use a newer version of a submodule, it can update it's submodule reference (see "Pull upstream changes to the submodule" on http://blog.jacius.info/git-submodule-cheat-sheet/). If this is done then running `git submodule update --recursive` is enough to bring all the submodules to the right revision.

**TL;DR:** I think 4215b8b3561840fc4959138a16ae0fe32fed8aa6 should be reverted.
